int16 default_value
int16 min_value
int16 max_value
